### ISSUE TITLE
Warn on old properties

### ISSUE THREAD

**Issue by philwebb** [spring-projects/spring-boot#11301]
It would be nice if we could warn if old properties are being used.

**Comment by Razi007** [comment]
Is it about using old annotations?

**Comment by snicoll** [comment]
No, it's about a project using an old property that we no longer bind. We can detect this since we have metadata for it so we're looking for a way to warn users on startup.

**Comment by Razi007** [comment]
i wan to pick this task guide me where can i start looking into the code

**Comment by philwebb** [comment]
@Razi007 This one might be a little hard to pick up, it's going to take some thought from our side. If you're interested in contributing something take a look at [issues labeled with "status: ideal-for-contribution"](https://github.com/spring-projects/spring-boot/labels/status%3A%20ideal-for-contribution) as they should be a bit easier to get started with. Thanks!

**Comment by snicoll** [comment]
@philwebb I've [hacked something](https://github.com/snicoll-scratches/spring-boot-config/tree/8548953566e61c3867a57369421de8655eb7bfd5/spring-boot-config-validator) that requires to read the metadata on startup so it has a dependency on `spring-boot-confguration-metadata` which itself brings json again. 

I didn't bother too much because I didn't  expect us to use that directly but this use case requires it. It is not clear where the code would live, having a compile scope dependency on something from `spring-boot-tools` makes me nervous. We'd also need to shade json so perhaps a dedicated module for this makes sense after all? :(

**Comment by snicoll** [comment]
Discussing this with phil, we'd create a `spring-boot-legacy-properties` module that can be added to a project. When that's the case, we'd analyse the `Environment` for deprecated properties.
